app/views/handlers/individual_response.py

```diff
@@ -491,13 +512,109 @@ def handle_post(self):
             self._update_section_status(CompletionStatus.INDIVIDUAL_RESPONSE_REQUESTED)
             return redirect(
                 url_for(
-                    "individual_response.get_individual_response_post_address_confirmation"
+                    "individual_response.get_individual_response_post_address_confirmation",
+                    return_to=self._return_to,
                 )
             )
 
         return redirect(
             url_for(
                 "individual_response.get_individual_response_how",
                 list_item_id=self._list_item_id,
+                return_to=self._return_to,
+            )
+        )
+
+
+class IndividualResponseWhoHandler(IndividualResponseHandler):
+    def __init__(self, schema, questionnaire_store, language, request_args, form_data):
+        self._list_name = schema.json.get("individual_response", {}).get("for_list")
+        list_model = questionnaire_store.list_store[self._list_name]
+        name_answers = []
+
+        for list_item in list_model.items:
+            if list_item != list_model.primary_person:
+                name_answers.append(
+                    (
+                        list_item,
+                        questionnaire_store.answer_store.get_answers_by_answer_id(
+                            ["first-name", "last-name"], list_item_id=list_item
+                        ),
+                    )
+                )
+
+        self.people_names = {
+            f"{name_answer[0].value} {name_answer[1].value}": list_item_id
+            for list_item_id, name_answer in name_answers
+        }
+        super().__init__(
+            self.block_definition,
+            schema,
+            questionnaire_store,
+            language,
+            request_args,
+            form_data,
+            request_args.get("list_item_id"),
+        )
+
+    @cached_property
+    def selected_option(self):
+        answer_id = self.rendered_block["question"]["answers"][0]["id"]
+        return self.form.get_data(answer_id)
+
+    @cached_property
+    def selected_list_item(self):
```

Comment:
Very minor, but I would absorb selected_option in here, I understood what select_list_item meant without the extra method and with it not being used anywhere else I feel it could sit as one.

Reply:
This is the approach adopted in the previous class, I think it's clearer to separate them.